README.md

```diff
@@ -1,27 +1,50 @@
-# frosty
+# `frosty`
+> reactive jamepad extension
+
 [![Codacy Badge](https://api.codacy.com/project/badge/Grade/a141c62a0d074580a10974a5ece29348)](https://www.codacy.com/app/delta-leonis/frosty?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=delta-leonis/yagl4j&amp;utm_campaign=Badge_Grade)
 [![CircleCI](https://circleci.com/gh/delta-leonis/algieba.svg?style=svg)](https://circleci.com/gh/delta-leonis/algieba)
 
 You'll need at least Java 1.8 ([jre](https://www.java.com/download/)
 /[jdk](http://www.oracle.com/technetwork/java/javase/downloads/index-jsp-138363.html))
 to run `frosty`.
 
-## Documentation
+## Dependency
+
+#### Maven
+
+```
+<dependency>
+    <groupId>io.leonis</groupId>
+    <artifactId>frosty</artifactId>
+    <version>0.0.1</version>
+</dependency>
+```
 
-The javadoc for the current code on `master` can be found [here](https://delta-leonis.github.io/frosty/)
+#### Gradle
 
-## Setup
+```
+compile 'io.leonis:frosty:0.0.1'
+```
+
+### Suppliers
+
+Since the data evolves as it is processed the type of the data container may change as well. In
+order to allow composition of these data containers over functions, these functions should be
+parametrized using interface instead of class types. Unfortunately Java's `Supplier` can only be
+implemented once due to type erasure; this is why most data types in `frosty` have a nested
+`Supplier` (or `SetSupplier`, `MappingSupplier`, e.a.) interface. This also allows for unnecessary
+coupling as an function that which uses a dpad can work with any controller with a dpad.
+
+## Documentation
 
-The following packages are required:
-- [GCC](https://gcc.gnu.org/install/binaries.html)
-- [SDL (Simple Directmedia Layer) 2](https://www.libsdl.org/download-2.0.php)
```

Comment:
why was this removed?

Reply:
Jamepad supports all platforms we want and building the sdl impl posed problems on at Mac os and possibly windows as well. Since jamepad is pretty complete I propose just to use that and not deal with jni.